validations/src/validations.rs

```diff
@@ -736,7 +736,15 @@ pub fn validate_block_transactions(
     let mut co_mt = ProgressiveMerkleTree::sha256();
     let mut commits_number = HashMap::new();
     let block_beacon = block.block_header.beacon;
+    let mut commit_hs = HashSet::new();
     for transaction in &block.txns.commit_txns {
+        if commit_hs.contains(&transaction.hash()) {
```

Comment:
Instead of checking the transaction hash, this should check `(dr_pointer, pkh)`, to ensure that there is only one commit per pkh per data request.

Reply:
Same with reveals and tallies.